python/plugins/processing/algs/qgis/Delaunay.py

```diff
@@ -111,10 +117,14 @@ def processAlgorithm(self, progress):
             attrs = []
             step = 0
             for index in indicies:
-                request = QgsFeatureRequest().setFilterFid(ptDict[ids[index]])
+                fid, n = ptDict[ids[index]]
+                request = QgsFeatureRequest().setFilterFid(fid)
```

Comment:
Could this be done with a single `setFilterFids` request instead?
See [this blog post for speeding up pyqgis queries](http://nyalldawson.net/2016/10/speeding-up-your-pyqgis-scripts/).